### PR TITLE
Homework7 git

### PR DESCRIPTION
**Pull request by Iliailineya**
Add hibernate to project

### CODE REVIEW

File: src/main/resources/app.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+db.driverClassName = com.mysql.cj.jdbc.Driver
+db.url = jdbc:mysql://localhost:3306/demo_db?
+db.username = root
+db.password = asSGs#1f$kdfj12sad8
```
Comment (YeaaaaH) [inline src/main/resources/app.properties]: don't share passwords in public

File: src/main/java/example/spring/service/AccountService.java [hunk 2]
```diff
@@ -1,82 +1,52 @@
 package example.spring.service;
 
-import example.spring.model.enams.Gender;
+import example.spring.exception.AccountNotFoundException;
 import example.spring.model.Account;
+import example.spring.repository.AccountsRepository;
+import jakarta.transaction.Transactional;
```
Comment (YeaaaaH) [inline src/main/java/example/spring/service/AccountService.java]: id you use @Transactional annotation it should be from spring package, but it's not needed now

File: pom.xml [hunk 3]
```diff
@@ -21,6 +21,13 @@
             <artifactId>spring-webmvc</artifactId>
             <version>6.0.9</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.springframework.data/spring-data-jpa -->
+        <dependency>
+            <groupId>org.springframework.data</groupId>
```
Comment (YeaaaaH) [inline pom.xml:28]: remove it, no need in spring-data at this moment

File: pom.xml [hunk 4]
```diff
@@ -55,6 +62,27 @@
             <artifactId>maven-war-plugin</artifactId>
             <version>3.4.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
+            <artifactId>spring-data-commons</artifactId>
```
Comment (YeaaaaH) [inline pom.xml]: remove it, no need in spring-data at this moment

File: src/main/java/example/spring/repository/AccountsRepository.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public Optional<Account> findById(Long id) {
+        Session currentSession = sessionFactory.getCurrentSession();
+        Account account = currentSession.get(Account.class, id);
+        return Optional.ofNullable(account);
+    }
+
+    public void updateAccount(Account account) {
+        Session currentSession = sessionFactory.getCurrentSession();
```
Comment (YeaaaaH) [inline src/main/java/example/spring/repository/AccountsRepository.java]: create separate session for each action: create, get, update, delete and don't forget to close it

File: src/main/java/example/spring/repository/AccountsRepository.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+public class AccountsRepository {
+    private final SessionFactory sessionFactory;
+
+    public AccountsRepository(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public Account save(Account account) {
+        Session currentSession = sessionFactory.getCurrentSession();
+        return currentSession.merge(account);
```
Comment (YeaaaaH) [inline src/main/java/example/spring/repository/AccountsRepository.java]: use save() and close it

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
need to fix comments

File: src/main/resources/app.properties (hunk 1 above)
Comment (Iliailineya) [reply src/main/resources/app.properties]: It was an accident

**Comment by Iliailineya** [comment]
Fixed

**Review by YeaaaaH — APPROVED** [review]

**Pull request merged**